streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -814,61 +814,72 @@ void runOnce() {
             addRecordsToTasks(records);
         }
 
+        // we can always let changelog reader to try restoring in order to initialize the changelogs;
+        // if there's no active restoring or standby updating it would not try to fetch any data
+        changelogReader.restore();
+
         // only try to initialize the assigned tasks
         // if the state is still in PARTITION_ASSIGNED after the poll call
         if (state == State.PARTITIONS_ASSIGNED) {
-            if (taskManager.initializeNewTasksAndCheckForCompletedRestoration()) {
+            if (taskManager.checkForCompletedRestoration()) {
```

Comment:
Here I have to break the `changelogReader#restore` out of the TaskManager so that 1) we call it every iteration no matter if we are restoring active or updating standby; 2) we only need to initialize tasks once.